drivers/scheduler/k8s/k8s.go

```diff
@@ -5055,89 +5130,126 @@ func (k *K8s) CreateCsiSnapsForVolumes(ctx *scheduler.Context, snapClass string)
 	return volSnapMap, nil
 }
 
-// CreateCsiSnapsAndValidate create CSI snapshot for Pure FA volumes and restore them to new PVCs, and validate the content
+// CSISnapshotTest create CSI snapshot for volumes and restore them to new PVCs, and validate the content
 // (Testrail cases C58775, 58091)
-func (k *K8s) CreateCsiSnapsAndValidate(ctx *scheduler.Context, snapClass string) error {
-	// This test will validate the content of the snapshot as opposed to just verify creation of snapshot.
-	timestamp := strings.Split(snapClass, "-")[0]
-	namespace := "fada-basic-snapshot-and-restore-test-" + timestamp
-	clientset, err := getKubeClient("")
+func (k *K8s) CSISnapshotTest(ctx *scheduler.Context, request scheduler.CSISnapshotRequest) error {
+	// This test will validate the content of the volume as opposed to just verify creation of volume.
+
+	pvcObj, err := k8sCore.GetPersistentVolumeClaim(request.OriginalPVCName, request.Namespace)
+	size := pvcObj.Spec.Resources.Requests[corev1.ResourceStorage]
+
 	if err != nil {
-		logrus.Errorf("Failed to get kube client for FADA snapshot and restore test: %s", err)
+		logrus.Errorf("Failed to retrieve PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
 		return err
 	}
-
-	// create a namespace for snapshot and restore test for FADA
-	nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
-	_, err = k8sCore.CreateNamespace(nsSpec)
+	originalStorageClass, err := k8sCore.GetStorageClassForPVC(pvcObj)
 	if err != nil {
-		logrus.Errorf("Failed to create new namespace %s FADA snapshot and restore test: %s", namespace, err)
+		logrus.Errorf("Failed to retrieve SC for PVC %s in namespace: %s : %s", request.OriginalPVCName, request.Namespace, err)
 		return err
 	}
+	storageClassName := originalStorageClass.Name
+	logrus.Infof("Procedding with SC %s", storageClassName)
 
-
-	originalPVCName := "fada-basic-pvc"
-	snapName := "fada-basic-pvc-snapshot" + timestamp
-	restoredPVCName := "fada-basic-pvc-restored" + timestamp
+	podsUsingPVC, err := k8sCore.GetPodsUsingPVC(pvcObj.GetName(), pvcObj.GetNamespace())
+	if err != nil {
+		logrus.Errorf("Failed to retrieve pods using PVC %s/%s", pvcObj.GetName(), pvcObj.GetNamespace())
+		return err
+	}
+	pod := podsUsingPVC[0]
+	mountPath, _ := pureutils.GetAppDataDir(podsUsingPVC[0].Namespace)
 	dirtyData := "thisIsJustSomeRandomText"
-	elasticsearchSC := "elasticsearch-sc"
-
-	logrus.Infof("Create original PVC: %s", originalPVCName)
-	originalPVCSpec := MakePVC(namespace, originalPVCName, elasticsearchSC)
-	originalPVC, err := k8sCore.CreatePersistentVolumeClaim(originalPVCSpec)
+	snapName := request.SnapName
+	for i :=0; i < 3; i++ {
+		data := fmt.Sprint(dirtyData, strconv.Itoa(int(time.Now().Unix())))
+		err = k.writeDataToPod(data, pod.GetName(), pod.GetNamespace(), mountPath)
```

Comment:
Are we intending to do it like:
* Write data
* Snap & verify
* Write data
* Snap & verify
* Write data
* Snap & verify
?
I kinda assumed we'd do:
* Write data
* Snap once
* Make three separate restores

Reply:
yea so this test is to write data snap and verify, write some more, and then verify

`I kinda assumed we'd do:
Write data
Snap once
Make three separate restores`
this is for NEWSTACK-573 and that will be a separate PR coming out later today( hopefully)